controllers/perconaserverformysql_controller.go

```diff
@@ -294,16 +330,31 @@ func reconcileReplicationPrimaryPod(
 
 func reconcileReplicationSemiSync(
 	ctx context.Context,
-	rdr client.Reader,
+	cl client.Reader,
 	cr *apiv2.PerconaServerForMySQL,
 ) error {
+	l := log.FromContext(ctx).WithName("reconcileReplicationPrimaryPod")
+
 	host := orchestrator.APIHost(orchestrator.ServiceName(cr))
 	primary, err := orchestrator.ClusterPrimary(ctx, host, cr.ClusterHint())
 	if err != nil {
-		return errors.Wrap(err, "get primary from orchestrator")
+		return errors.Wrap(err, "get cluster primary")
+	}
+
+	primaryHost := primary.Hostname()
+	if host == "" {
```

Comment:
how can `host` be empty here? if it's empty I believe line 301 should fail. Also `orcHost` is a more descriptive name

Reply:
sometimes it is empty while alias is set